rust/ruby-rbs/src/lib.rs

```diff
@@ -40,6 +40,41 @@ pub fn parse(rbs_code: &[u8]) -> Result<*mut rbs_signature_t, String> {
     }
 }
 
+pub struct NodeListIter {
+    parser: *mut rbs_parser_t,
+    current: *mut rbs_node_list_node_t,
+}
+
+impl Iterator for NodeListIter {
+    type Item = Node;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.current.is_null() {
+            None
+        } else {
+            let node = unsafe { Node::new(self.parser, (*self.current).node) };
+            self.current = unsafe { (*self.current).next };
```

Comment:
Would this work? If not, no worries
```suggestion
            let pointer_data = unsafe { *self.current };
            let node = Node::new(self.parser, pointer_data.node);
            self.current = pointer_data.next;
```

Reply:
I still need to wrap `Node::new` with an `unsafe` block, but I do like your suggestion as it avoids dereferencing `self.current` twice